### ISSUE TITLE
std.Variant.opIndex returns a Variant

### ISSUE THREAD

**Issue by Geod24** [s-ludwig/std_data_json#15]
This isn't really a problem in itself, but makes the interface harder to use than it should be:

```D
unittest {
    auto txt = `{ "name": "foo", "lang": "D" }`;
    auto json = parseJSONValue(txt);

    // Pass
    assert(json["name"].get!JSONValue.get!string == "foo");

   // The assert(0) will get triggered
    try {
        assert(json["lang"].get!string == "D");
    } catch (VariantException)
        assert(0, "VariantException");
}
```

**Comment by s-ludwig** [comment]
Fixed with the switch to `TaggedAlgebraic`:
```
unittest {
    auto txt = `{ "name": "foo", "lang": "D" }`;
    auto json = parseJSONValue(txt);

    // Pass
    assert(json["name"] == "foo");
    assert(json["lang"] == "D");
}
```

**Comment by Geod24** [comment]
:+1: 
One issue closed, one opened (#17) :wink:
